src/lib/utils/evm/approve.ts

```diff
@@ -0,0 +1,51 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { CONTRACT_INTERACTION } from '@shapeshiftoss/chain-adapters'
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import type { ApproveInputWithWallet } from 'react-queries/queries/mutations'
+
+import { assertGetEvmChainAdapter, buildAndBroadcast, getApproveContractData, getFees } from '.'
+
+export const approve = async ({
```

Comment:
Not react-query specific anymore, just a good ol' async fn 🎉

Reply:
So good! Purity 🤝